### PR TITLE
Propagate HADOOP_PROXY_USER on launching Flink engine

### PR DESCRIPTION
**Pull request by dongkelun**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala [hunk 1]
```diff
@@ -137,6 +137,9 @@ trait ProcBuilder {
 
     val envs = pb.environment()
     envs.putAll(env.asJava)
+    if (shortName.equals("flink")) {
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:140]: we can move this to flink-proc-builder？

Comment (dongkelun) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:140]: There is childProcEnv in the history, but it is removed from this PR [https://github.com/apache/incubator-kyuubi/pull/2572](https://github.com/apache/incubator-kyuubi/pull/2572). Can I add it again?

Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:140]: maybe we can put it into `pb.environment()` directly

Comment (dongkelun) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala]: Would it be better to move HADOOP_PROXY_USER to conf of FlinkProcessBuilder?

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
cc @yanghua @SteNicholas, do you think it's a bug fix or new feature? should we merge it into branch-1.6?

**Review by pan3793 — APPROVED** [review]

**Review by yanghua — APPROVED** [review]
+1

**Comment by pan3793** [comment]
Thanks, merging to master

**Comment by pan3793** [comment]
@dongkelun the `HADOOP_PROXY_USER` has side-effects and causes #4762 on non-kerberized env, any thoughts?

**Comment by pan3793** [comment]
@dongkelun have done some investigation, seems Flink does not support proxy user, so there is no ideal solution IMO now.

I will simply revert this patch from branch-1.7(master already removed it in b315123a6b6dfa7b03a5ab7875856bdfd4e0eaed), as a workaround, users can pass the env vars to the engine by `kyuubi.engineEnv.HADOOP_PROXY_USER=<>` to achieve this capability

**Pull request closed without merging**